subprojects/base-services/src/main/java/org/gradle/internal/operations/DescribableBuildOperation.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.internal.operations;
+
+/**
+ * Interface used by build operations which also provide a display name
+ * or an operation descriptor.
+ *
+ * @param <T> the type of the operation descriptor
+ */
+public interface DescribableBuildOperation<T> extends BuildOperation {
```

Comment:
Also note that my first attempt was just to have a `Builder BuildOperationDetails.from(BuildOperation)` but it creates a cyclic dependency between packages.